src/utilities/fused_increment.jl

```diff
@@ -35,39 +35,80 @@ function fused_increment end
 
 @inline fused_increment(u, dt, sc, tend, v) = fused_increment(u, dt, sc, tend, v, get_S(sc))
 
-# ij case (S::NTuple{2})
-@generated function fused_increment(u, dt, sc::T, tend, ::Val{i}, ::NTuple{2}) where {i, S, T <: SparseCoeffs{S}}
-    i ≤ 1 && return :(u)
-    terms = []
-    for j in 1:(i - 1)
-        zero_coeff(T, i, j) && continue
-        push!(terms, :(Base.Broadcast.broadcasted(*, dt * sc[$i, $j], tend[$j])))
+# =================================================== ij case (S::NTuple{2})
+# recursion: _rfused_increment_j always returns a Tuple
+@inline _rfused_increment_ij(js::Tuple{}, i, u, dt, sc::SparseCoeffs, tend) = ()
```

Comment:
One downside of recursive loop unrolling (in contrast to generative unrolling) is that the compiler stops generating type-stable code after a certain recursion depth. You might need to disable the recursion limit for this function (and also for `_rfused_increment_j`) in order to avoid allocations when using tableaus with a lot of stages, like `ARK548L2SA2`.

Reply:
In order to avoid dissabling a bunch of recursion limits, I ended up doing all of my unrolling in `ClimaCore.MatrixFields` with the generated functions in `ClimaCore.Utilities.UnrolledFunctions` (some of which are just imported from `Unrolled.jl`). You could potentially use one of those here instead.

Reply:
I looked at JET, and the issue is that the `1:s` loop is not unrolled, so `i` is not compile-time known. I'm pretty sure that I can fix it, I just figured that I'd try and get #224 merged first